src/arcaflow_plugin_sdk/schema.py

```diff
@@ -2674,13 +2674,30 @@ def _insert_discriminator(
         discriminated_object: typing.Dict[str, typing.Any],
         discriminator_value: str,
     ) -> typing.Dict[str, typing.Any]:
+        """This function adds a member type's discriminator field as a
+        property with a constant value equal to its discriminated value.
+        The discriminator field is moved to the zeroth index of the list
+        of required fields in a data packet.
+
+        :param discriminated_object: A dictionary in the form of a json fragment
+            of an object's schema in the scope of this OneOf's union.
+        :param discriminator_value: The value that represents the given object in
+            its OneOf's union.
+        :returns: A dictionary in the form of a json fragment of an object's schema
+            in the scope of this OneOf's union updated to reflect its membership
+            in this OneOf's union.
+        """
```

Comment:
There's a subtlety with regard to docstrings:  the first line/sentence is supposed to be a brief summary of the function; then there should be a line break/blank-line; then there should be any additional explanation, and then parameter descriptions, etc.

There are tools which extract the first line (up to the line break) and report it; so, having a "first line" which consists of two sentences spanning four lines is probably not what we want.  Something more like this might be better:
> Add a discriminator field as a property of a member type
>
>  This function adds a member type's discriminator field as a
        property with a constant value equal to its discriminated value.
        The discriminator field is moved to the zeroth index of the list
        of required fields in a data packet.
>
> [Parameter descriptions, etc.]

Ditto line 2842, et al.

Reply:
did it